configuration/yaml.md

```diff
@@ -0,0 +1,644 @@
+Variables can be defined in multiple places, resolved in the following order of precedence:
+
+1. Inline `vars` in `!include` directives
+1. Global `variables` defined in the main file
+1. Local `variables` defined in the [included file](#including-other-files)
+
+Variables defined inline with `!include` take precedence over all others and apply only to that include context.
+This allows included files to be reused with different values, making them behave like parameterized templates.
+
+Example:
+
+```yaml
+version: 1
+
+variables:
+  broker: mqtt:broker:main
+  device_id1: sensor01
+  location1: Kitchen
+  device_id2: sensor02
+  location2: Living Room
+
+things:
+  mqtt:topic:${device_id1}:
+    label: "${location1} Sensor"
+    bridge: ${broker}
+    config:
+      availabilityTopic: "${device_id1}/status"
+
+  mqtt:topic:${device_id2}:
+    label: "${location2} Sensor"
+    bridge: ${broker}
+    config:
+      availabilityTopic: "${device_id2}/status"
+```
+
+The resulting configuration will resolve to:
+
+```yaml
+version: 1
+
+things:
+  mqtt:topic:sensor01:
+    label: "Kitchen Sensor"
+    bridge: mqtt:broker:main
+    config:
+      availabilityTopic: "sensor01/status"
+
+  mqtt:topic:sensor02:
+    label: "Living Room Sensor"
+    bridge: mqtt:broker:main
+    config:
+      availabilityTopic: "sensor02/status"
+```
+
+#### Special Variables
+
+openHAB introduces a set of special variables that are automatically available during YAML preprocessing.
+
+Available Special Variables:
+
+| Variable        | Description                                                                 |
+|:----------------|:----------------------------------------------------------------------------|
+| `__FILE__`      | Absolute path to the current file, e.g. `/path/to/file.inc.yaml`            |
+| `__FILE_NAME__` | The filename portion without the extension or leading path, e.g. `file.inc` |
+| `__FILE_EXT__`  | The file extension portion of the file name, e.g. `yaml`                    |
+| `__DIRECTORY__` | The directory portion of the current file, e.g. `/path/to`                  |
+| `__DIR__`       | Alias for `__DIRECTORY__`.                                                  |
+
+These variables can be interpolated just like regular ones using `${...}` syntax.
+These will be useful when combined with the `!include` directive described below.
+
+### Including Other Files
+
+openHAB supports including external YAML files to facilitate modular, reusable, and maintainable configurations.
+This is especially useful for templating, creating device packages, or separating concerns across multiple files.
+
+#### Syntax Options
+
+Two syntaxes are supported for the `!include` directive:
+
+##### Simple Syntax
+
+```yaml
+keyname: !include filename.inc.yaml
+```
+
+Use this when no variable overrides are needed.
+
+##### Full Syntax
+
+```yaml
+keyname: !include
+  file: filename.inc.yaml
+  vars:
+    var1: value1
+    var2: value2
+```
+
+Or compact form:
```

Comment:
I think this looks a bit strange. Maybe `##### Compact form` could be used instead?

Reply:
The compact form is essentially still a full syntax, so it shouldn't have its own section. It's just a YAML shorthand thing.

Reply:
I have added spaces around the emdash.



> I understand that, it's just that when the font ends up being much larger for the compact form, I find it "confusing": When you use 5 #'s the font gets quite small. It's a markdown quirk I guess, the font sizes used depends on the rendering and is somewhat out of your control. I just thought it looked strange, so I made a note. Maybe there's no "better" way to do it.

I see your point. I've combined them into the same code fence.